categorize_notify/program.py

```diff
@@ -0,0 +1,162 @@
+
+
+def _process_email(gmail, message_id: str):
+    message = gmail.messages().get(userId="me", id=message_id).execute()
+    email_content = _parse_email(message)
+    if email_content:
+        channel = _categorize_email(email_content, SLACK_CHANNELS)
+        if channel:
+            client = slack.slack_client(SLACK_CONNECTION_NAME)
+            client.chat_postMessage(channel=channel, text=email_content)
+
+        # Add label to email
+        label_id = _get_label_id(gmail, channel)
+        if not label_id:
+            created_label = _create_label(gmail, channel)
+            label_id = created_label["id"]
+        body = {"addLabelIds": [label_id]}
+        gmail.messages().modify(userId="me", id=message_id, body=body).execute()
+
+
+def _get_latest_message_ids(gmail, new_email_count: int):
+    """Get the latest email message_id from the user's inbox.
+    Args:
+        gmail: An authorized Gmail API service instance.
+        new_email_count: The number of new email messages to retrieve.
+    Returns: A list of email message_id strings.
+    """
+    results = gmail.messages().list(userId="me", maxResults=new_email_count).execute()
+
+    return [msg["id"] for msg in results.get("messages", [])]
+
+
+def _parse_email(message: dict):
+    """Parse provided email.
+    Args:
+        message: The email message to parse
+    Returns: The email body as a string.
+    """
+    payload = message["payload"]
+    for part in payload.get("parts", []):
+        if part["mimeType"] == "text/plain":
+            return base64.urlsafe_b64decode(part["body"]["data"]).decode("utf-8")
+
+
+def _create_label(gmail, label_name: str) -> dict:
+    """Create a new label in the user's gmail account.
+    Args:
+        gmail: An authorized Gmail API service instance.
+        label_name: The name of the label to be created.
+    Returns: The created label as a dictionary.
+    https://developers.google.com/gmail/api/reference/rest/v1/users.labels#Label
+    """
+    label = {
+        "labelListVisibility": "labelShow",
+        "messageListVisibility": "show",
+        "name": label_name,
+    }
+    created_label = gmail.labels().create(userId="me", body=label).execute()
+    print(f"Label created: {created_label['name']}")
+    return created_label
+
+
+def _get_label_id(gmail, label_name: str) -> str:
+    """Get the label_id for a label with the provided name if it exists.
+    Args:
+        gmail: An authorized Gmail API service instance.
+        label_name: The name of the label to retrieve the id for.
+    """
+    labels_response = gmail.labels().list(userId="me").execute()
+    labels = labels_response.get("labels", [])
+    for label in labels:
+        if label["name"] == label_name:
+            return label["id"]
+    return None
+
+
+def _get_message_count(gmail) -> int:
+    """Returns The total number of messages in the user's inbox."""
+    profile = gmail.getProfile(userId="me").execute()
+    return profile["messagesTotal"]
```

Comment:
You can combine these 2 lines into one, and not eliminate this function altogether.

Reply:
Done.